src/Coordination/KeeperStateMachine.cpp

```diff
@@ -523,9 +520,8 @@ void KeeperStateMachine::processReadRequest(const KeeperStorage::RequestForSessi
         true /*check_acl*/,
         true /*is_local*/);
     for (const auto & response : responses)
-        if (!responses_queue.push(response) && !responses_queue.isFinished())
-            throw Exception(
-                ErrorCodes::SYSTEM_ERROR, "Could not push response with session id {} into responses queue", response.session_id);
+        if (!responses_queue.push(response))
```

Comment:
Why you have removed all this `isFinished` logic?

Reply:
because `push` can return `false` only if it's finished, i.e. with the previous check the throw was dead code.
Don't think a comment is needed because it should be an API-defined thing for queue.